### PR TITLE
Update MiniCart.php to fix price total amount on page load prices are including tax

### LINKED ISSUE #7830: MiniCart.php - 'wc-block-mini-cart__amount' not displaying correct amount when including tax on product prices on initial page load.
## Describe the bug

On initial page load the mini-cart is not showing the correct total when you have items already in the cart and you have set tax to be inclusive with prices. Looks to be showing the total amount exclusive of tax. Does however update correctly when hovering over the mini-cart which I'm guessing is handled via a AJAX request.

## To reproduce

Steps to reproduce the behaviour:

1. Enable Have product prices inclusive of tax
2. Add Mini cart block to widget which is places in the header of the template
3. Add items to cart
4. Reload the page.
5. Shows cart total exclusive of tax until you interact with the mini-cart.

## Expected behaviour

Expected to see Total amount inclusive of tax. Not exclusive on page load.

Line 305 around function 'get_cart_price_markup' should probably include the following after the '$cart_contents_total' variable

`if ( $cart->display_prices_including_tax() ) { $cart_contents_total += $cart->get_subtotal_tax(); }`

## Screenshots

[https://i.ibb.co/KNFYv6n/cart-error.gif](https://i.ibb.co/KNFYv6n/cart-error.gif)

### WordPress (please complete the following information):

-   WordPress version: 6.1.1
-   WooCommerce version: 7.1.0
-   Theme: Storefront: 4.2.0

### Desktop (please complete the following information):

-   Browser: Chrome - Latest

### PR DESCRIPTION
**Pull request by lukeoregan88**
Fixes #7830

Added an update to the '$cart_contents_total' variable on the 'get_cart_price_markup' function if $cart->display_prices_including_tax is true. Currently displaying wrong amount total on page load if items are already in the basket.

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Go to WooCommerce > Settings and set _Display prices during cart and checkout_ to _Including tax_.
Settings screenshot
2. Create a post or page and add the Mini Cart block.
3. In the frontend, hover the Mini Cart block with the mouse and notice the price doesn't update ([before](https://user-images.githubusercontent.com/3616980/206694129-c16fcea4-8ac4-4bd7-a72a-221946c3ef08.webm) and [after](https://user-images.githubusercontent.com/3616980/206694060-eee1cccf-1ebd-435d-995d-23caca715918.webm)).

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Fix wrong Mini Cart amount displayed when displaying prices including taxes.

### CODE REVIEW

File: src/BlockTypes/MiniCart.php [hunk 1]
```diff
@@ -303,6 +303,11 @@ protected function get_cart_price_markup( $attributes ) {
 		$cart_controller     = $this->get_cart_controller();
 		$cart                = $cart_controller->get_cart_instance();
 		$cart_contents_total = $cart->get_subtotal();
+
```
Comment (Aljullu) [inline src/BlockTypes/MiniCart.php]: There are a couple of PHP linting errors because of having white spaces at the end of the file. Here and in line 310.

**Review by Aljullu — COMMENTED** [review]
Thanks for contributing this fix, @lukeoregan88! Could you write some testing steps to allow us to reproduce the issue and see how this PR fixes it? It would be good to know what are your settings in the _Tax_ tab, specifically:

* Prices entered with tax 
* Display prices in the shop 
* Display prices during cart and checkout 

Thanks in advance!

**Review by lukeoregan88 — COMMENTED** [review]
Removed white spaces

**Comment by lukeoregan88** [comment]
So I have an example of a website I've just built...

It was recently updated to Woocommerce 7.1.1

1. Go to the website [https://ottomanoils.com](https://ottomanoils.com)
2. Add [Spiced Orange Beard Oil/](https://ottomanoils.com/product/spiced-orange-infusion-beard-oil/) to the basket. The price including TAX should be £25.00 (UK TAX is 20%)
3. Refresh or head back to the Homepage
4. Check Mini Cart from the header of the website. Initial Load says £20.83.
5. On mini-cart interaction price updates to £25.00

Mini-cart popout does show the correct total basket price of £25.00. This is also correct for the basket and checkout pages.

This code will fix it. It was actually code taken from an early version of Woocommerce but for some reason, it was removed from version 7? Probably because new blocks were introduced?

I think probably a better solution would be to amend the AJAX function for the mini-cart. On interaction (hover) of the mini-cart the basket updates to include the 20% tax. Maybe this should be extended to include on-page load also? This would also be handy for sites that utilise cache plugins.

File: src/BlockTypes/MiniCart.php [hunk 2]
```diff
@@ -303,6 +303,9 @@ protected function get_cart_price_markup( $attributes ) {
...
+		if($cart->display_prices_including_tax() ) { $cart_contents_total += $cart->get_subtotal_tax();}
```
Comment (Aljullu) [inline src/BlockTypes/MiniCart.php]: The formatting of this `if` needs to be updated:
```suggestion
		if ( $cart->display_prices_including_tax() ) {
			$cart_contents_total += $cart->get_subtotal_tax();
		}
```

**Review by Aljullu — COMMENTED** [review]
Thanks for updating this PR, @lukeoregan88! This is testing well. Heads-up that I updated your original PR description to include in-detail testing steps.

Before merging, there is one last code formatting change that is needed. :point_down:

**Review by lukeoregan88 — COMMENTED** [review]
Fixed IF

**Comment by Aljullu** [comment]
Thanks, @lukeoregan88! I just merged this PR and it will be released soon in WC Blocks and, after that, in WC core. :tada:

**Comment by lukeoregan88** [comment]
Thank you very much @Aljullu . This was my first ever pull request and fix for a public repo. I usually only fix in-house private projects. Thanks for the support. I look forward to seeing the fix in future updates of the WC core 🙌🏻. Have a great weekend and Merry Christmas 🎄.

**Pull request merged**